src/wp-includes/blocks.php

```diff
@@ -1020,11 +1043,11 @@ function set_ignored_hooked_blocks_metadata( &$parsed_anchor_block, $relative_po
  *                                                       or pattern that the blocks belong to.
  * @param callable                        $callback      A function that will be called for each block to generate
  *                                                       the markup for a given list of blocks that are hooked to it.
- *                                                       Default: 'insert_hooked_blocks'.
+ *                                                       Default: 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata'.
  * @return callable A function that returns the serialized markup for the given block,
  *                  including the markup for any hooked blocks before it.
  */
-function make_before_block_visitor( $hooked_blocks, $context, $callback = 'insert_hooked_blocks' ) {
+function make_before_block_visitor( $hooked_blocks, $context, $callback = 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata' ) {
```

Comment:
I'm not sure we can change the defaults here, since it's going to change the behavior of the visitor when called without an explicit `$callback` argument, with respect to its previous behavior in 6.4 and 6.5 😕 

We might need to ask around. @gziolo what's your stance on this?

If we have to keep the default, then we need to update the callsites instead. (Fortunately, there aren't too many, and it probably makes sense to make the argument explicit everywhere anyway.)

Reply:
Hmm I see your concern. I figured that we would have to do one or the other and since the function is marked as `@private` I'm not sure what the difference would be between updating the default vs updating the call sites given all of these functions using it are marked explicitly for internal use.

FWIW I think updating the call sites would be better anyway since it's more explicit but interested to know more about what I mentioned above either way.

Reply:
> I figured that we would have to do one or the other and since the function is marked as `@private` I'm not sure what the difference would be between updating the default vs updating the call sites given all of these functions using it are marked explicitly for internal use too.

Indeed. I seem to recall that with Core's fairly strict back-compat policy, none of these PHPDocs annotations really override the need to keep the function backwards compatible 😬 

(IIRC, the only time it's more relaxed is for "actually `private` class methods.)

Reply:
What will happen if the default value changes? This function shouldn’t be used by extenders as it is marked as private, but if they use it regardless their code shouldn’t break. You can search the plugins directory to confirm it is used and contact authors if necessary. The dev note will also help mitigate the impact.

Reply:
>What will happen if the default value changes?

If the default value changes to my proposed value, the blocks are inserted as they currently are, the only difference is that we perform the additional step of adding the `ignoredHookedBlocks` meta to the template markup on 'reading' rather than just doing it on 'writing' (ie when saving to the database).

>This function shouldn’t be used by extenders as it is marked as private, but if they use it regardless their code shouldn’t break

This is my opinion also, which is why I assumed it was 'safe' to update the default value.

>You can search the plugins directory to confirm it is used and contact authors if necessary. The dev note will also help mitigate the impact.

Good to know!

@ockham with all of the above said, are you still of the opinion that we should only update the call sites _or_ the default value?

Reply:
I updated the call sites @ockham as per our conversation https://github.com/WordPress/wordpress-develop/pull/6358/commits/4cb44f28d99b6a60d583247970b56c42c8eb953a

Reply:
Thank you, Tom!